pilot/pkg/xds/workloadentry.go

```diff
@@ -121,6 +130,49 @@ func (sg *InternalGen) QueueUnregisterWorkload(proxy *model.Proxy) {
 	}, features.WorkloadEntryCleanupGracePeriod)
 }
 
+// UpdateWorkloadEntryHealth updates the associated WorkloadEntries health status
+// based on the corresponding health check performed by istio-agent.
+func (sg *InternalGen) UpdateWorkloadEntryHealth(proxy *model.Proxy, event HealthEvent) {
+	if !features.WorkloadEntryHealthChecks {
+		return
+	}
+	// we assume that the workload entry exists
+	// if auto registration does not exist, try looking
+	// up in NodeMetadata
+	entryName := autoregisteredWorkloadEntryName(proxy)
+	if entryName == "" {
+		// todo make this env var a constant somewhere
+		name, ok := proxy.Metadata.ProxyConfig.ProxyMetadata["WORKLOAD_ENTRY_NAME"]
```

Comment:
The value should be prefixed with ISTIO_META_. Then it will be put in node metadata, and you can access by `proxy.metadata.WorkloadEntryName` (once you add the field)

Reply:
if we are doing non-auto registration cases later this will be in a different pr